packages/server/src/oauth/token.ts

```diff
@@ -462,7 +462,25 @@ async function parseClientAssertion(clientAssertiontype: string, clientAssertion
 
   try {
     await jwtVerify(clientAssertion, JWKS, verifyOptions);
-  } catch (err) {
+  } catch (error: any) {
+    // There are some edge cases where there are multiple matching JWKS
+    // and we need to iterate throught the JWKSMultipleMatchingKeys error
+    // and return the first verified match
+    if (error?.code === 'ERR_JWKS_MULTIPLE_MATCHING_KEYS') {
```

Comment:
https://github.com/panva/jose/blob/main/docs/functions/jwks_remote.createRemoteJWKSet.md

This shows the logic for multiple matching keys